ethereum/contracts/relayer/wormholeRelayer/WormholeRelayerDelivery.sol

```diff
@@ -0,0 +1,645 @@
+                RETURNDATA_TRUNCATION_THRESHOLD
+            );
+
+            Gas postGas = Gas.wrap(gasleft());
+
+            unchecked {
+                gasUsed = (preGas - postGas).min(gasLimit);
+            }
+        }
+
+        if (success) {
+            targetRevertDataTruncated = new bytes(0);
+            status = uint8(DeliveryStatus.SUCCESS);
+
+            ForwardInstruction[] storage forwardInstructions = getForwardInstructions();
+
+            if (forwardInstructions.length > 0) {
+                //Calculate the amount of the transaction fee to refund to the user (multiply the maximum refund by the
+                //  fraction of gas unused)
+                LocalNative transactionFeeRefundAmount = (gasLimit - gasUsed).toWei(
+                    evmInstruction.targetChainRefundPerGasUnused
+                ).asLocalNative();
+                emitForward(gasUsed, transactionFeeRefundAmount, forwardInstructions);
+                status = uint8(DeliveryStatus.FORWARD_REQUEST_SUCCESS);
+            }
+        } else {
+            // Note that forward instructions should always be empty in this case.
+            status = uint8(DeliveryStatus.RECEIVER_FAILURE);
+        }
+    }
+
+    /**
+     * - Checks if enough funds were passed into a forward
+     * - Increases the 'extraReceiverValue' of the first forward in order to use all of the funds
+     * - Publishes the DeliveryInstruction
+     * - Pays the relayer's reward address to deliver the forward
+     *
+     * @param transactionFeeRefundAmount amount of maxTransactionFee that was unused
+     * @param forwardInstructions An array of structs containing information about the user's forward
+     *     request(s)
+     */
+    function emitForward(
+        Gas gasUsed,
+        LocalNative transactionFeeRefundAmount,
+        ForwardInstruction[] storage forwardInstructions
+    ) private {
+        LocalNative wormholeMessageFee = getWormholeMessageFee();
+
+        //Decode send requests and aggregate fee and payment
+        DeliveryInstruction[] memory instructions =
+            new DeliveryInstruction[](forwardInstructions.length);
+
+        LocalNative totalMsgValue;
+        LocalNative totalFee;
+        for (uint256 i = 0; i < forwardInstructions.length;) {
+            unchecked {
+                totalMsgValue = totalMsgValue + forwardInstructions[i].msgValue;
+            }
+            instructions[i] =
+                (forwardInstructions[i].encodedInstruction).decodeDeliveryInstruction();
+            totalFee = totalFee + forwardInstructions[i].deliveryPrice
+                + forwardInstructions[i].paymentForExtraReceiverValue + wormholeMessageFee;
+            unchecked {
+                ++i;
+            }
+        }
+
+        //Combine refund amount with any additional funds which were passed in to the forward as
+        //  msg.value and check that enough funds were passed into the forward 
+        LocalNative fundsForForward;
```

Comment:
`fundsForForward` can replace `totalMsgValue` on line 427 to reduce local variable count and simplify the code. For example:

`LocalNative fundsForForward = transactionFeeRefundAmount;`